libraries/chain/database_worker_objects.cpp

```diff
@@ -209,18 +209,32 @@ namespace golos { namespace chain {
 
         const auto now = head_block_time();
 
-        const auto& idx = get_index<worker_techspec_index>().indices().get<by_post>();
+        const auto& idx = get_index<worker_techspec_index, by_post>();
         for (auto itr = idx.begin(); itr != idx.end(); itr++) {
+            if (itr->state != worker_techspec_state::created) {
+                continue;
+            }
+
             const auto& wto_post = get_comment(itr->post);
             if (wto_post.created + mprops.worker_techspec_approve_term_sec > now) {
                 break;
             }
 
-            if (itr->state != worker_techspec_state::created) {
+            close_worker_techspec(*itr, worker_techspec_state::closed_by_witnesses);
```

Comment:
Techspec is not closed by witnesses.
It is closed by timeout.

Reply:
@afalaleev Do we need special state - closed_by_timeout?